README.md

```diff
@@ -1,6 +1,37 @@
 # ansible-role-cloudstack-init
 
-Initialize a host for cloudstack environment
+Initialize a host for cloudstack environment.
+
+This role configure VMs to:
+
+* install ssh key pair (`keypair` in `deployVirtualMachine`)
+* update root password if available
+* run `user-data` (`userdata` in `deployVirtualMachine`)
+
+upon the next boot.
+
+Also, the role does:
+
+* clean up VM
+
+## Notes for FreeBSD users
+
+`user-data` is processed in the following order.
+
+If the first two bytes are '#!', make it executable and run it.
+
+If the first two bytes are '>/', the first line contains a path for the
+rest of the file to be written into.
+
+If the first three bytes are '>>/', the first line contains a path for the
+rest of the file to be appended into.
+
+Otherwise, hope this is an archive containing more files for us to process.
```

Comment:
i do not understand what is `an archive` and `more files` without reading the `templates/cs_configinit.j2`.
more explanation about the archive is preferable.
it seems `an archive` must be a tar file with/without compression and `more files` are files described above
and will be processed recursively, which should be described here.